pkg/kubelet/prober/manager.go

```diff
@@ -74,36 +80,63 @@ type manager struct {
 func NewManager(
 	defaultProbePeriod time.Duration,
 	statusManager status.Manager,
-	prober Prober) Manager {
+	readinessManager results.Manager,
+	livenessManager results.Manager,
+	runner kubecontainer.ContainerCommandRunner,
+	refManager *kubecontainer.RefManager,
+	recorder record.EventRecorder) Manager {
+	prober := newProber(runner, refManager, recorder)
 	return &manager{
 		defaultProbePeriod: defaultProbePeriod,
 		statusManager:      statusManager,
 		prober:             prober,
-		readinessCache:     results.NewManager(),
-		readinessProbes:    make(map[containerPath]*worker),
+		readinessManager:   readinessManager,
+		livenessManager:    livenessManager,
+		workers:            make(map[probeKey]*worker),
 	}
 }
 
-// Key uniquely identifying containers
-type containerPath struct {
+// Key uniquely identifying container probes
+type probeKey struct {
 	podUID        types.UID
 	containerName string
+	probeType     probeType
 }
 
+// Type of probe (readiness or liveness)
+type probeType bool
+
+const (
```

Comment:
nit: I understand bool is more economical here, but using int seems more extensible and safer (e.g.,  accidentally used in an if-else condition)
```
const (
    liveness probeType = iota // 0
    readiness                 // 1
)
```